include/vcpkg/base/expected.h

```diff
@@ -134,13 +134,13 @@ namespace vcpkg
         // Constructors are intentionally implicit
 
         ExpectedT(const S& s, ExpectedRightTag = {}) : m_s(s) { }
-        template<class = std::enable_if<!std::is_reference<S>::value>>
+        template<std::enable_if_t<!std::is_reference<S>::value, int> = 0>
         ExpectedT(S&& s, ExpectedRightTag = {}) : m_s(std::move(s))
         {
         }
 
         ExpectedT(const T& t, ExpectedLeftTag = {}) : m_t(t) { }
-        template<class = std::enable_if<!std::is_reference<T>::value>>
+        template<std::enable_if_t<!std::is_reference<S>::value, int> = 0>
```

Comment:
`T` -> `S` copypaste error?

Reply:
🤦

Reply:
https://github.com/microsoft/vcpkg-tool/pull/470

Reply:
Interestingly, if I fix this that doesn't compile because you can't use enable_if on a non-function-parameter-type like that :sigh:

Reply:
I wonder if something like `ExpectedT(const T&& t, ExpectedLeftTag = {})` is more appropriate here.

Reply:
That wouldn't work because the goal is to move-from

Reply:
Is this constraint even needed? https://godbolt.org/z/48n34n4sj